plugins/auth-react/src/hooks/useCookieAuthRefresh/useCookieAuthRefresh.tsx

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useEffect, useState, useCallback } from 'react';
+import {
+  discoveryApiRef,
+  fetchApiRef,
+  useApi,
+} from '@backstage/core-plugin-api';
+import { useAsync, useMountEffect } from '@react-hookz/web';
+import { ResponseError } from '@backstage/errors';
+
+type CookieAuthRefreshMessage = MessageEvent<{
+  action: string;
+  payload: {
+    expiresAt: string;
+  };
+}>;
+
+/**
+ * @public
+ * The options for the {@link useCookieAuthRefresh} hook.
+ */
+export type CookieAuthRefreshOptions = {
+  // The plugin ID to used for discovering the API origin
+  pluginId: string;
+  // The path to used for calling the refresh cookie endpoint, default to '/cookie'
+  path?: string;
+};
+
+/**
+ * @public
+ * A hook that will refresh the cookie when it is about to expire.
+ * @remarks
+ * This hook expects a `BroadcastChannel` to be available in the global scope.
```

Comment:
The code could fairly easily  be written in such a way that it tolerates it being not defined. It's kind of a "bonus" function after all.

Reply:
After your comment, I had an idea of replacing the usage of the broadcast channel with storage api... wdyt?